src/AST/PropertyExtensions.cs

```diff
@@ -14,7 +14,7 @@ public static bool IsInRefTypeAndBackedByValueClassField(this Property p)
             type = type ?? p.Field.Type;
 
             Class decl;
-            return type.TryGetClass(out decl) && decl.IsValueType;
+            return type.TryGetClass(out decl) && (decl.IsValueType || decl.IsStruct);
```

Comment:
I first thought IsValueType and IsStruct were mean to be in different contexts. But now yes they seem to be same. Will move it.